src/core/Directus/Database/TableGateway/BaseTableGateway.php

```diff
@@ -461,13 +462,13 @@ public function updateRecordByArray(array $recordData)
 
         $replace = true;
         if ($collectionName === SchemaManager::COLLECTION_FILES && static::$container) {
```

Comment:
Same as above. This kind of exception really shouldn't be this deep down the abstraction layer.

We have both `FilesService.php` and `AssetsService.php` to handle everything that's specific to saving files and retrieving assets. This should move into there